crates/supervisor/storage/src/log/mdbx.rs

```diff
@@ -0,0 +1,363 @@
+
+use crate::{
+    error::StorageError,
+    log::{LogStorageReader, LogStorageWriter},
+    models::{BlockRefs, LogEntries},
+};
+use kona_protocol::BlockInfo;
+use kona_supervisor_types::Log;
+use reth_db_api::{
+    cursor::{DbCursorRO, DbDupCursorRO, DbDupCursorRW},
+    transaction::{DbTx, DbTxMut},
+};
+use std::fmt::Debug;
+use tracing::{debug, error, warn};
+
+/// A log storage that wraps a transactional reference to the MDBX backend.
+#[derive(Debug)]
+#[allow(dead_code)]
+pub(crate) struct MdbxLogStorage<'tx, TX> {
+    tx: &'tx TX,
+}
+
+/// Internal constructor and setup methods for [`MdbxLogStorage`].
+impl<'tx, TX> MdbxLogStorage<'tx, TX> {
+    #[allow(dead_code)]
+    pub(crate) const fn new(tx: &'tx TX) -> Self {
+        Self { tx }
+    }
+}
+
+/// Implements the [`LogStorageWriter`] trait
+impl<TX> LogStorageWriter for MdbxLogStorage<'_, TX>
+where
+    TX: DbTxMut,
+{
+    type Error = StorageError;
+
+    fn store_block_logs(&self, block: &BlockInfo, logs: Vec<Log>) -> Result<(), Self::Error> {
+        debug!(target: "supervisor_storage", block_number = block.number, "Storing logs",);
+        let tx = self.tx;
+        if let Err(e) = tx.put::<BlockRefs>(block.number, (*block).into()) {
+            error!(
+                target: "supervisor_storage",
+                block_number = block.number,
+                error = ?e,
+                "Failed to write block"
+            );
+            return Err(StorageError::Database(e));
+        }
+
+        let mut cursor = tx.cursor_dup_write::<LogEntries>().map_err(|e| {
+            error!(target: "supervisor_storage", error = ?e, "Failed to get dup cursor");
+            StorageError::Database(e)
+        })?;
+
+        for log in logs {
+            if let Err(e) = cursor.append_dup(block.number, log.into()) {
+                error!(
+                    target: "supervisor_storage",
+                    block_number = block.number,
+                    error = ?e,
+                    "Failed to append logs"
+                );
+                return Err(StorageError::Database(e));
+            }
+        }
+        Ok(())
+    }
+}
+/// Implements the [`LogStorageReader`] trait
+impl<TX> LogStorageReader for MdbxLogStorage<'_, TX>
+where
+    TX: DbTx,
+{
+    type Error = StorageError;
+    fn get_block(&self, block_number: u64) -> Result<BlockInfo, Self::Error> {
+        debug!(target: "supervisor_storage", block_number = block_number, "Fetching block");
+
+        let block_option = self.tx.get::<BlockRefs>(block_number).map_err(|e| {
+            error!(
+                target: "supervisor_storage",
+                block_number = block_number,
+                error = ?e,
+                "Failed to read block",
+            );
+            StorageError::Database(e)
+        })?;
+
+        let block = block_option.ok_or_else(|| {
+            warn!(target: "supervisor_storage", block_number = block_number, "Block not found");
```

Comment:
```suggestion
            warn!(target: "supervisor_storage", block_number, "Block not found");
```
you can just do this and it will use the variable name as key name